swift_scality_backend/diskfile.py

```diff
@@ -56,22 +56,16 @@ def __init__(self, filesystem, name):
         headers = {
             'transfer-encoding': 'chunked'
         }
-        self.logger.debug("DiskFileWriter for %s initialized", self.safe_path)
+        self.logger.debug("DiskFileWriter for %s initialized", self._name)
 
-        (ipaddr, port) = self._filesystem.sproxyd_hosts.next()
-        with swift.common.exceptions.ConnectionTimeout(filesystem.conn_timeout):
-            self._conn = swift.common.bufferedhttp.http_connect_raw(
-                ipaddr, port, 'PUT', self.safe_path, headers)
+        self._conn, self._release_conn = filesystem.get_http_conn_for_put(
+            self._name, headers)
 
     def __repr__(self):
         ret = 'DiskFileWriter(filesystem=%r, object_name=%r)'
         return ret % (self._filesystem, self._name)
 
-    logger = property(lambda self: self._filesystem.logger)
-
-    @property
-    def safe_path(self):
-        return self._filesystem.base_path + urllib.quote(self._name)
+    logger = property(lambda self: self._filesystem._logger)
```

Comment:
Cause the way we pass the logger(s) is bad. It needs a proper fix, subject to another PR.